handlers/handlers.go

```diff
@@ -0,0 +1,55 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/config"
+	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/mapper"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// ClientError is an interface that can be used to retrieve the status code if a client has errored
+type ClientError interface {
+	Error() string
+	Code() int
+}
+
+func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
+	status := http.StatusInternalServerError
+	if err, ok := err.(ClientError); ok {
+		if err.Code() == http.StatusNotFound {
```

Comment:
This seems counter-productive. If you're going to allow retrieving the statusCode you shouldn't restrict it to one specific statusCode. I think this should be
```
if err, ok := err.(ClientError); ok {
  status = err.Code()
}
```